hazelcast/src/main/java/com/hazelcast/client/impl/protocol/codec/builtin/ListCNFixedSizeCodec.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl.protocol.codec.builtin;
+
+import com.hazelcast.client.impl.protocol.ClientMessage;
+import com.hazelcast.internal.util.BiTuple;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import static com.hazelcast.client.impl.protocol.codec.builtin.FixedSizeTypesCodec.BYTE_SIZE_IN_BYTES;
+import static com.hazelcast.client.impl.protocol.codec.builtin.FixedSizeTypesCodec.INT_SIZE_IN_BYTES;
+import static com.hazelcast.client.impl.protocol.codec.builtin.FixedSizeTypesCodec.decodeByte;
+import static com.hazelcast.client.impl.protocol.codec.builtin.FixedSizeTypesCodec.decodeInt;
+import static com.hazelcast.client.impl.protocol.codec.builtin.FixedSizeTypesCodec.encodeByte;
+import static com.hazelcast.client.impl.protocol.codec.builtin.FixedSizeTypesCodec.encodeInt;
+
+/**
+ * A common codec that is used to serialize a collections with a fixed-size nullable elements.
+ */
+final class ListCNFixedSizeCodec {
+
+    private static final byte TYPE_NULL_ONLY = 1;
+    private static final byte TYPE_NOT_NULL_ONLY = 2;
+    private static final byte TYPE_MIXED = 3;
+
+    private static final int ITEMS_PER_BITMASK = 8;
+
+    private static final int HEADER_SIZE = BYTE_SIZE_IN_BYTES + INT_SIZE_IN_BYTES;
+
+    private ListCNFixedSizeCodec() {
+    }
+
+    public static <T> void encode(
+        ClientMessage clientMessage,
+        Iterable<T> items,
+        int itemSizeInBytes,
+        EncodeFunction<T> encodeFunction
+    ) {
+        BiTuple<Integer, Integer> itemCounts = countItems(items);
+
+        int totalItemCount = itemCounts.element1;
```

Comment:
Why not just pass `Collections` instead of `Iterable` and have the `size`. This way countItems will just return single int .